### PR TITLE
Refactored some util functions and some minor tweaks for saving tracks

### PR DESCRIPTION
**Pull request by Broxzier**
- Saving a ride will no longer display ".TD6" at the end of the filename - this gets added automatically now.
- Having dots in the middle of the filename are no issue, the full filename will be displayed in the track list.
- Changed some funcitons in util.c to not do while loops manually to find the last dot (`strrchr` finds it, or returns NULL if nothing is found), and added comments to them.

### CODE REVIEW

File: src/util/util.c [hunk 1]
```diff
@@ -55,62 +55,48 @@ bool filename_valid_characters(const utf8 *filename)
 
 const char *path_get_filename(const utf8 *path)
 {
-	const char *result, *ch;
+	// Find last slash or backslash in the path
+	char *filename = strrchr(path, '/');
```
Comment (janisozaur) [inline src/util/util.c]: why not `platform_get_path_separator`?

Comment (Broxzier) [inline src/util/util.c]: Oh nice, didn't know about that function yet. It's being used now.

**Comment by IntelOrca** [comment]
save scenario also uses path_set_extension,

**Comment by Broxzier** [comment]
Don't merge this just yet, I'm changing some places where `path_get_filename` is used where not needed.

**Comment by Broxzier** [comment]
Alright, I'm done now. :)

**Comment by Broxzier** [comment]
There's one thing that's different for now: the extension `.TD6` will not be displayed in the window when saving a track design (by design), but this window, which is still the default one, only warns the player if there is a file with the same filename if it includes the extension. Additionally it does warn the user when saving a track including the extension, though then it will actually save it as `filename.TD6.TD6`.

**Comment by Broxzier** [comment]
Does this still need testing?

**Pull request merged**